cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataHttpHandlerTest.java

```diff
@@ -377,6 +378,30 @@ public void testInvalidTags() throws IOException {
     Assert.assertEquals(400, addTags(application, tags).getResponseCode());
   }
 
+  @Test
+  public void testDeletedProgramHandlerStage() throws Exception {
+    deploy(WordCountApp.class, Constants.Gateway.API_VERSION_3_TOKEN, TEST_NAMESPACE1);
+    Id.Program program = Id.Program.from(TEST_NAMESPACE1, "WordCountApp", ProgramType.FLOW, "WordCountFlow");
+
+    // Set some properties metadata
+    Map<String, String> flowProperties = ImmutableMap.of("sKey", "sValue", "sK", "sV");
+    addProperties(program, flowProperties);
+
+    // Get properties
+    Map<String, String> properties = getProperties(program);
+    Assert.assertEquals(2, properties.size());
+
+    // Deploy WordCount App without Flow program.
+    deploy(WordCountMinusFlowApp.class, Constants.Gateway.API_VERSION_3_TOKEN, TEST_NAMESPACE1);
+
+    // Get properties from deleted (flow) program - should return 404
+    getPropertiesFromInvalidEntity(program);
+
+    //Delete the App after stopping the flow
```

Comment:
Yeah, I do not need to start/stop the flow. Just need the updated app to contain no Flow program.